service/controller/app/v1/resource/releasemigration/create.go

```diff
@@ -65,26 +85,121 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return nil
 	}
 
+	// If Helm v2 release configmap had been deleted and Helm v3 release secret was created,
+	// It means helm v3 release migration is finished.
+	if !hasConfigMap && hasSecret {
+		err = r.uncordonChart(ctx, cc.Clients.K8s.G8sClient())
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
 	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("no pending migration for release %#q", key.ReleaseName(cr)))
 
 	return nil
 }
 
-func (r *Resource) hasHelmV2ConfigMaps(ctx context.Context, k8sClient kubernetes.Interface, releaseName string) (bool, error) {
+func (r *Resource) cordonChart(ctx context.Context, g8sClient versioned.Interface) error {
+	lo := metav1.ListOptions{
+		LabelSelector: "app notin (chart-operator)",
+	}
+	charts, err := g8sClient.ApplicationV1alpha1().Charts("giantswarm").List(lo)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("cordoning %d charts", len(charts.Items)))
+
+	cordonReason := replaceToEscape(fmt.Sprintf("%s/%s", annotation.ChartOperatorPrefix, annotation.CordonReason))
+	cordonUntil := replaceToEscape(fmt.Sprintf("%s/%s", annotation.ChartOperatorPrefix, annotation.CordonUntil))
+	patches := []patch{
+		{
+			Op:    "add",
+			Path:  fmt.Sprintf("/metadata/annotations/%s", cordonReason),
+			Value: "Migrating to helm 3",
+		},
+		{
+			Op:    "add",
+			Path:  fmt.Sprintf("/metadata/annotations/%s", cordonUntil),
+			Value: key.CordonUntilDate(),
+		},
+	}
```

Comment:
What if there are no annotations? 

We had that problem in chart-operator. I think it would be safer to have that logic. https://github.com/giantswarm/chart-operator/blob/ae3cb2f4c26821f66aa47199bdbc64855580d402/service/controller/chart/resource/release/resource.go#L121-L127